iree/test/e2e/linalg_ext_ops/BUILD

```diff
@@ -42,12 +44,14 @@ iree_check_single_backend_test_suite(
     srcs = enforce_glob(
         # keep sorted
         [
+            "linalg_transform.mlir",
             "reverse.mlir",
             "scan.mlir",
         ],
         include = ["*.mlir"],
-        exclude = [
-        ],
+        # linalg_transform_spec is a an MLIR file that specifies a
+        # transformation, it needs to be included as data.
+        exclude = ["linalg_transform_spec.mlir"],
```

Comment:
I think this may need to pass `-linalg-transform-file-name=${CMAKE_CURRENT_SOURCE_DIR}/linalg_transform_spec.mlir` in via `COMPILER_FLAGS`, and the https://github.com/google/iree/blob/main/iree/compiler/Translation/test/linalg_transform_spec.mlir file should also be moved.

However, since that's complicated... maybe it would be better to have a lit test for `-linalg-transform-file-name` in `iree/compiler/` then only test the results of what _that_ generates in `iree/test/e2e/`? Do we expect IREE users to be passing that flag with mlir files as part of their own compiler invocations, or is it mostly a tool for compiler developers? (I hope the latter)

Reply:
Realized this as soon as the test failed. I dont know how to tell bazel (and then from there to CMake through bazel_to_cmake) to specify this extra MLIR file in the command line. Any pointers on that? I uploaded changes I made so far, but obviously they dont work.

> Do we expect IREE users to be passing that flag with mlir files as part of their own compiler invocations, or is it mostly a tool for compiler developers? (I hope the latter)

Of course this is just a straw man. This is never something that is supposed to a "user-level" feature. More for experimentation + development. The lit test you mention already exists (https://github.com/google/iree/blob/main/iree/compiler/Codegen/LLVMCPU/test/linalg_transform.mlir) . This is more for e2e testing.

Reply:
Also see that this isnt enough. Need to plumb this through all the CMake macros. @nicolasvasilache do we really want this e2e test?

Reply:
A few comments, there are 3 things here: the existence of an integration test, the implementation of the test itself and the deeper reason why this uses a file.

Re. having e2e integration tests. I think we do: this is pushing into the dreaded tensor + parallelism abstractions territory. In fact, note the comment in the test:
```
  // util.do_not_optimize on output to prevent fusing in the same dispatch
  // region which would be subject to racy tensor semantics.
  // Forcing different dispatches forces flow.dispatch.tensor.load which is
  // actually side-effecting.
```
We should prob. have a negative test there too to capture this, I have not yet investigated further, having instead invested in building up my velocity within the project.

Re. implementation of the test. There is value in connecting tools to run e2e with out relying on orthogonal testing infrastructure, that is non-standard and does not seem to support the options LIT supports. In fact, during my bringup of the tests into IREE (this PR and previous ones), I found one of the most time-consuming aspect was to find the proper way to write tests and to connect IREE tools. The fact that the proposed solution is to drop the LIT `// RUN` command is not appealing to me.

Separately, the fact that this is "complicated" is kind of a red flag to me: the ongoing evolution of MLIR is towards less static precompiled C++ (compiler compile-time) and more dynamically evaluated PDL / PDLL (compiler runtime), at least for controlling what gets applied where. The source of this PDL IR / PDLL language is "files". Granted in the future we'll need `include` and loader mechanisms, but let's not put the cart before the wheels. 

Re. deeper reason to use a file. Stepping back a little, we can dig deeper into why this particular test uses a file: the MLIR to drive the transformation is self-contained. The limitation we uncovered at the time was related to transporting the information to the right place at the right time in the compiler flow: basically during the various passes ran before we touch codegen, the MLIR transform module gets separated and couldn't be recovered easily and applied. Solving this involves deeper contracts/agreements between IREE assumptions and the layering of a new dialect that starts dipping a toe in the PDL pool. I don't think it makes sense to start digging into this until this new line of work has been demonstrated within IREE (i.e. by showing some close to peak performance on some small model).

To avoid opening this layering can of worms, a simple solution is to specify the transformation separately. It was introduced [here](https://github.com/google/iree-llvm-sandbox/commit/30c98c9e374249beb8ae961c7bf780398a281c4e). 
This does not exhibit any complication and I expect we'd want to allow similar separation of concerns in all our code.

IMO, a simple solution is to start an Integrations directory like we have in core, where tests are based on LIT, that can depend on any iree-tools and exhibit a low degree of surprise and compose in similar ways we expect in core. I have found this to be lacking and it would have saved me days of "getting started" investigations. We can address all the other issues separately as they get closer to blocking.

Happy to hop on a VC to answer specific questions or concerns.

Reply:
I dont see the relevance of this w.r.t to the issue at hand. The issue is that these tests need to move somewhere else. Just need to figure out how to get the cmake/bazel setup here to use the mlir file provided. Given that there are lit tests that check each step, could avoid the extra work needed here (cause I guess thats on my plate, and I really would rather not do it) to get the e2e working in CI. It would still work locally.

Reply:
Thanks Scott! I was looking for something like this. I'd push against more such e2e tests. We need to figure out a better way of handling this without using such a flag. One e2e test to ensure the light stays on with that path, and rest with lit tests should be good enough.

Reply:
Okay, if we're not going to add dozens of these style of tests, then a new hostonly test suite somewhere under iree/test/e2e/ using iree_lit_test may be the best solution for now. The general solution with check test, Bazel, and CMake is certainly possible, it'll just take a bit of plumbing.

Reply:
ok, thats good. Ill try that....